test/schema-feature.test.js

```diff
@@ -1289,3 +1290,320 @@ test('setSchemaController per instance', t => {
 
   fastify.ready(err => { t.error(err) })
 })
+
+test('setSchemaController: Inherits correctly parent schemas with a customized validator instance', async t => {
+  t.plan(5)
+  const customAjv = new Ajv({ coerceTypes: false })
+  const server = Fastify()
+  const someSchema = {
+    $id: 'some',
+    type: 'array',
+    items: {
+      type: 'string'
+    }
+  }
+  const errorResponseSchema = {
+    $id: 'error_response',
+    type: 'object',
+    properties: {
+      statusCode: {
+        type: 'integer'
+      },
+      message: {
+        type: 'string'
+      }
+    }
+  }
+
+  // these schemas must be added before calling setSchemaController()
```

Comment:
This needs to be documented. Shouldn't we consider a start execution flow to avoid this kind of issue in `next`?

Reply:
>This needs to be documented. 

I agree on this being documented, maybe for the current version?

>Shouldn't we consider a start execution flow to avoid this kind of issue in next?

I'm not so sure I'm following you. I think is not so possible in the current state due to the fact the only place where you can really wait for all the Schemas being added is when registering the routes. Maybe when `#addSchema` is executed we can also try somehow to pass the schemas down to the factory? But that wil require quite few changes in the current logic.

But maybe @Eomm has a better idea

Reply:
It was an idea for fastify v4, not for the current version as it adds a lot of changes in the whole process.

Reply:
The schema controllers are built during the startup phase actually.

You can move the `addSchema` calls before the inject 👍🏽

Reply:
Oh, yeah, you're right, right here: 
https://github.com/fastify/fastify/blob/6a0403319148163fe6f826aba16e33e46fdb88d4/lib/route.js?_pjax=%23repo-content-pjax-container#L274

On next still behaves the same so there's no issue at all. Don't recall why I added such comments, let me delete them to avoid confusion 👍